### PR TITLE
Bitbucket fixes

### PR DESCRIPTION
**Pull request by gdetrez**
This fixes a couple of issues I noticed with the bitbucket service in the latest version:

* bugwarrior was only pulling a small number of issues from bitbucket
* when using `only_if_assigned`, no issues were imported

### CODE REVIEW

File: bugwarrior/services/bitbucket.py [hunk 1]
```diff
@@ -157,12 +173,13 @@ def validate_config(cls, config, target):
         IssueService.validate_config(config, target)
 
     def fetch_issues(self, tag):
-        response = self.get_data('/repositories/%s/issues/' % (tag))
-        return [(tag, issue) for issue in response['values']]
+        response = self.get_collection('/repositories/%s/issues/' % (tag))
+        for issue in response:
+            yield (tag, issue)
```
Comment (ralphbean) [inline bugwarrior/services/bitbucket.py]: Why employ a generator here when we just collapse it with a ``list()`` below?

Also, why a generator here and not in ``fetch_pull_requests``.  I don't mind using one over the other, just so long as its consistent.

File: bugwarrior/services/bitbucket.py [hunk 2]
```diff
@@ -204,8 +204,10 @@ def get_annotations2(self, tag, issue, issue_obj, url):
         )
 
     def get_owner(self, issue):
-        tag, issue = issue
-        return issue.get('responsible', {}).get('username', None)
+        _, issue = issue
+        assignee = issue.get('assignee', None)
+        if assignee is not None:
+            return assignee.get('username', None)
```
Comment (ryneeverett) [inline bugwarrior/services/bitbucket.py]: Is this a documented change?

Comment (gdetrez) [inline bugwarrior/services/bitbucket.py]: Yes, it's part of the new API

File: bugwarrior/services/bitbucket.py (hunk 1 above)
Comment (gdetrez) [inline bugwarrior/services/bitbucket.py]: Hum... good point ;-)

File: bugwarrior/services/bitbucket.py (hunk 2 above)
Comment (ryneeverett) [inline bugwarrior/services/bitbucket.py]: IMO the oneliner is more readable but I'll leave that to your discretion.

Comment (ryneeverett) [inline bugwarrior/services/bitbucket.py]: Link to the new API docs for the curious: https://confluence.atlassian.com/bitbucket/version-2-423626329.html#Version2-Issues

Comment (gdetrez) [inline bugwarrior/services/bitbucket.py]: @ryneeverett the behavior is slightly different: the one-liner crashes if `'assignee'` is a valid key in `issue` but the value is `None` (which does happen in the data returned by bitbucket).

File: bugwarrior/services/bitbucket.py (hunk 1 above)
Comment (gdetrez) [inline bugwarrior/services/bitbucket.py]: I think I planned to use only iterators but it turned out to require too much changes to the method `issues`. I reverted to a list in the last commit.

File: bugwarrior/services/bitbucket.py (hunk 2 above)
Comment (ryneeverett) [inline bugwarrior/services/bitbucket.py]: I get it now, thanks!

**Comment by ryneeverett** [comment]
:+1: to merge

**Comment by ralphbean** [comment]
Go Team!  :rocket:

**Pull request merged**